test/distributed/fsdp/test_hsdp_dtensor_state_dict.py

```diff
@@ -0,0 +1,251 @@
+# Owner(s): ["oncall: distributed"]
```

Comment:
I'll add a TODO to consolidate the test cases. I'd like to keep two tests for now temporarily since we are still actively developing, and it will be easier to test.